### PR TITLE
#958 reduce database disk space usage

### PR DESCRIPTION
**Pull request by wsbrenk**
1. changes proposed in this pull request:
 
   - fixes issue #958 
   
   - if not fixing an existing issue, comments explaining the purpose of the PR should be provided)
 
 
2. changelog and release_notes ...

 - [x] have been updated
 - [ ] do not require update

3. [Optional] suggested person to review this PR @___

### CODE REVIEW

File: src/main/java/core/db/DBUpdater.java [hunk 1]
```diff
@@ -72,6 +72,12 @@ void updateDB(int DBVersion) {
 	}
 
 	private void updateDBv600(int dbVersion) throws SQLException {
+		// reduce data base file's disk space
```
Comment (akasolace) [inline src/main/java/core/db/DBUpdater.java:75]: as I have lost my DB I can't check but I imagine those 2 tables are by far the largest. You know you can look into SYSTEM_TABLESTATS to see usage per table ?

File: src/main/java/core/db/BasicsTable.java [hunk 2]
```diff
@@ -41,7 +41,6 @@ protected void initColumns() {
 	@Override
 	protected String[] getCreateIndexStatement() {
 		return new String[] {
-			"CREATE INDEX IBASICS_1 ON " + getTableName() + "(HRF_ID)",
```
[the remark this answers is not in the document]
Comment (wsbrenk) [reply src/main/java/core/db/BasicsTable.java:44]: from hsqldb doc: 

Indexes should not duplicate the columns of PRIMARY KEY, UNIQUE or FOREIGN key constraints as each of these constraints creates an index automatically.

File: src/main/java/core/db/DBUpdater.java [hunk 3]
```diff
@@ -87,8 +93,15 @@ private void updateDBv600(int dbVersion) throws SQLException {
 		if (!tableExists(NtTeamTable.TABLENAME)) {
 			dbManager.getTable(NtTeamTable.TABLENAME).createTable();
 		}
-		dbManager.getTable(TeamTable.TABLENAME).tryDropIndex("ITEAM_1");
```
[the remark this answers is not in the document]
Comment (wsbrenk) [reply src/main/java/core/db/DBUpdater.java:96]: see above

File: src/main/java/core/db/DBUpdater.java (hunk 1 above)
Comment (wsbrenk) [reply src/main/java/core/db/DBUpdater.java:75]: i found largest ids for those two tables in my database

**Comment by akasolace** [comment]
Applying this patch on your data has which impact on the database size on disk? 
Also does it really close #958 ? I mean does it explain why the db size increased at that time ?

**Comment by wsbrenk** [comment]
@akasolace my disk space was reduced by a factor of 8. I'm not quiet sure what the time was, when size increased, but upgrade process HO4 to HO5 is not so unlikely, since we changed a lot and maybe this yields to massive fragmentations.

**Pull request merged**